web/server.go

```diff
@@ -0,0 +1,217 @@
+//go:generate statik -src=../assets
+
+// Package web Cozy Stack API.
+//
+// Cozy is a personal platform as a service with a focus on data.
+//
+// Terms Of Service:
+//
+// there are no TOS at this moment, use at your own risk we take no responsibility
+//
+//     Schemes: https
+//     Host: localhost
+//     BasePath: /
+//     Version: 0.0.1
+//     License: AGPL-3.0 https://opensource.org/licenses/agpl-3.0
+//     Contact: Bruno Michel <bruno@cozycloud.cc> https://cozy.io/
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+// swagger:meta
+package web
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/cozy/cozy-stack/config"
+	"github.com/cozy/cozy-stack/couchdb"
+	"github.com/cozy/cozy-stack/web/jsonapi"
+	_ "github.com/cozy/cozy-stack/web/statik" // Generated file with the packed assets
+	"github.com/labstack/echo"
+	"github.com/rakyll/statik/fs"
+)
+
+var templatesList = []string{
+	"login.html",
+}
+
+// Config represents the configuration
+type Config struct {
+	Router    *echo.Echo
+	Domain    string
+	Assets    string
+	ServeApps func(c echo.Context, domain, slug string) error
+}
+
+type renderer struct {
+	t *template.Template
+	h http.Handler
+}
+
+func (r *renderer) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
+	return r.t.ExecuteTemplate(w, name, data)
+}
+
+func (r *renderer) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	r.h.ServeHTTP(res, req)
+}
+
+func createRenderer(conf *Config) (*renderer, error) {
+	// By default, use the assets packed in the binary
+	if conf.Assets != "" {
+		if !path.IsAbs(conf.Assets) {
+			return nil, fmt.Errorf("path %s is not absolute", conf.Assets)
```

Comment:
Please, can we keep the option to use a relative path for assets? It's handy when working on the CSS (`go run main.go serve --assets ./assets`)